torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -178,10 +178,10 @@ static c10::optional<c10::ScalarType> InferExpectedScalarType(const Node* n) {
           } else {
             typesFromTensors.emplace_back(scalar_type);
           }
-        } else if (
-            auto scalar_type =
-                input->type()->cast<TensorType>()->scalarType()) {
-          typesFromTensors.emplace_back(*scalar_type);
+        } else if (auto tensor_type = input->type()->cast<TensorType>()) {
```

Comment:
Sometimes some nodes (scripting) have output List rather than Tensor, so it triggers seg fault here. This is a check for that. We may need revisit.